core/src/impl/Kokkos_Combined_Reducer.hpp

```diff
@@ -294,13 +294,24 @@ struct CombinedReducerImpl<std::integer_sequence<size_t, Idxs...>, Space,
     return m_value_view;
   }
 
-  KOKKOS_FUNCTION
+  template <int Idx, class View>
+  constexpr static void write_one_value_back(
+      View const& view, typename View::value_type const& value) noexcept {
+    if (Kokkos::SpaceAccessibility<typename View::memory_space,
+                                   Space>::assignable)
+      view() = value;
+    else
+      Kokkos::deep_copy(view, value);
```

Comment:
Is that a new fencing bug when an execution space instance was passed?

Reply:
Let me try to improve this a little.